### DISCUSSION TITLE
Error Handling in Csla.Blazor.ViewModel during SaveAsync

### DISCUSSION THREAD

**Discussion by ecaldentey** [MarimerLLC/csla#4268] [Questions]
Hello,

I am using the Csla.Blazor.ViewModel class to interact with my components in the UI, for example:
Inject protected Csla.Blazor.ViewModel<BusinessDealProductService> _vmProductService { get; set; }

When I call Save using the class like this: 
await vmBizdeal.SaveAsync();

If an error occurs during the database update process, the SaveAsync method does not throw an exception. The ViewModel class has an Exception property, but I'm not clear on the correct way to use this information.

How can I detect at the UI level if an error occurs when executing the SaveAsync method?

**Comment by StefanOssendorf** [comment]
Hi. 
You have to check the `Exception` and/or `ViewModelErrorText` property to see if any error happend during Save/Refresh.
And with that you can react to that. But I'm currently not using Blazor so maybe someone else can give better guidance.

Hth

**Reply by rockfordlhotka** [reply]
That is correct.

I usually bind one or both of those properties to the UI so the user (or developer) can see the error.

Alternately, after you've awaited the save call, you can check the `Exception` property to see if there was an exception if you need to handle something in code.
